app/models/order_detail.rb

```diff
@@ -283,21 +283,15 @@ def can_be_viewed_by?(user)
   scope :purchased, -> { joins(:order).merge(Order.purchased) }
 
   scope :pending, -> { joins(:order).where(state: %w(new inprocess)).purchased }
-  scope :confirmed_reservations, -> { reservations.joins(:order).includes(:reservation).purchased }
-
-  scope :upcoming_reservations, lambda {
-                                  confirmed_reservations
-                                    .where("reservations.reserve_end_at > ? AND reservations.actual_start_at IS NULL", Time.zone.now)
-                                    .order("reservations.reserve_start_at ASC")
-                                }
-
-  scope :in_progress_reservations, lambda {
-    confirmed_reservations
-      .merge(Reservation.relay_in_progress)
-      .order("reservations.reserve_start_at ASC")
-  }
 
-  scope :all_reservations, -> { confirmed_reservations.order("reservations.reserve_start_at DESC") }
+  scope :with_reservation, -> { purchased.joins(:reservation).order("reservations.reserve_start_at DESC") }
+  scope :with_upcoming_reservation, lambda {
+    non_canceled
+      .with_reservation
+      .where(reservations: { actual_start_at: nil })
+      .merge(Reservation.ends_in_the_future)
+  }
+  scope :with_in_progress_reservation, -> { non_canceled.with_reservation.merge(Reservation.relay_in_progress) }
```

Comment:
I can't remember off the top of my head--do these get ordered elsewhere in the code before they're displayed in the view? I just want to make sure the default ordering in the view(s) hasn't changed.

Reply:
The `with_reservation` scope just above, which these two scopes call, includes the order!

Previously, all three scopes included the exact same call to order, so I refactored it into the scope that is used as a base for these.

Reply:
Ah gotcha. Makes sense!